models/layers.py

```diff
@@ -846,61 +846,57 @@ def forward(self, x, Z, H, W):  # x: [1, 521280, 384], Z: 8, H: 181, W: 360
         return output
 
 
-class PatchRecoveryUpper(nn.Module):
-    """Patch recovery operation for wind power generation based on upper atmospheric variables.
-    This class processes only the atmospheric data (excluding surface data) to predict wind capacity.
-    Output dimensions: [1, 721, 1440], similar to the surface-level recovery.
-    """
-
+class PatchRecoveryPowerAll(nn.Module):
     def __init__(self, dim):
         super().__init__()
         """Patch recovery operation"""
+        # Hear we use two transposed convolutions to recover data
         self.patch_size = (2, 4, 4)
         self.dim = dim  # 384
-        # Receive 384 input images, project down to 32 output images
+
         self.conv = nn.Conv1d(in_channels=dim, out_channels=32, kernel_size=1, stride=1)
 
     def forward(self, x, Z, H, W):  # x: [1, 521280, 384], Z: 8, H: 181, W: 360
+        # The inverse operation of the patch embedding operation, patch_size = (2, 4, 4) as in the original paper
```

Comment:
Whilst this isn't your function, this might be one case where I would consider expanding the docstrings and describing what each parameter is. That would make it much easier to understand what you changed in order to enable patch recovery for the power values - i.e. why you slice differently etc.
If this is too much work then don't do it - but some form of explanation about what each dimension is could help

Reply:
Good point. I've just done that in another branch, where I tested out different Patch recoveries: https://github.com/EliasKng/pangu-pytorch/commit/6e92bb1787f059cc294896cf1e4df10a2a92efb8

I plan on merging this branch next week.